src/renderer/dx/DxFontInfo.cpp

```diff
@@ -0,0 +1,391 @@
+// - We'll try to fall back to Consolas with the given weight/stretch/style first,
+//   then try Consolas again with normal weight/stretch/style,
+//   and if nothing works, then we'll throw an error.
+// Arguments:
+// - dwriteFactory - The DWrite factory to use
+// - localeName - Locale to search for appropriate fonts
+// Return Value:
+// - Smart pointer holding interface reference for queryable font data.
+[[nodiscard]] Microsoft::WRL::ComPtr<IDWriteFontFace1> DxFontInfo::ResolveFontFaceWithFallback(gsl::not_null<IDWriteFactory1*> dwriteFactory,
+                                                                                               std::wstring& localeName)
+{
+    // First attempt to find exactly what the user asked for.
+    _didFallback = false;
+    auto face = _FindFontFace(dwriteFactory, localeName);
+
+    if (!face)
+    {
+        // If we missed, try looking a little more by trimming the last word off the requested family name a few times.
+        // Quite often, folks are specifying weights or something in the familyName and it causes failed resolution and
+        // an unexpected error dialog. We theoretically could detect the weight words and convert them, but this
+        // is the quick fix for the majority scenario.
+        // The long/full fix is backlogged to GH#9744
+        // Also this doesn't count as a fallback because we don't want to annoy folks with the warning dialog over
+        // this resolution.
+        while (!face && !_familyName.empty())
+        {
+            const auto lastSpace = _familyName.find_last_of(UNICODE_SPACE);
+
+            // value is unsigned and npos will be greater than size.
+            // if we didn't find anything to trim, leave.
+            if (lastSpace >= _familyName.size())
+            {
+                break;
+            }
+
+            // trim string down to just before the found space
+            // (space found at 6... trim from 0 for 6 length will give us 0-5 as the new string)
+            _familyName = _familyName.substr(0, lastSpace);
+
+            // Try to find it with the shortened family name
+            face = _FindFontFace(dwriteFactory, localeName);
+        }
+
+        // Alright, if our quick shot at trimming didn't work either...
+        // move onto looking up a font from our hardcoded list of fonts
+        // that should really always be available.
+        if (!face)
+        {
+            for (const auto fallbackFace : FALLBACK_FONT_FACES)
+            {
+                _familyName = fallbackFace;
+                face = _FindFontFace(dwriteFactory, localeName);
+
+                if (face)
+                {
+                    _didFallback = true;
+                    break;
+                }
+
+                _familyName = fallbackFace;
+                _weight = DWRITE_FONT_WEIGHT_NORMAL;
+                _stretch = DWRITE_FONT_STRETCH_NORMAL;
+                _style = DWRITE_FONT_STYLE_NORMAL;

```

Comment:
The second time this loops around it'll not call `_FindFontFace` with the weight/stretch/style the user asked for.

Reply:
I simply moved the original code here as part of this grand refactor. I didn’t modify any of these, just to make it easy to review (for Michael specifically, who wrote this in #9734)